t/15_qesap_azure.t

```diff
@@ -146,4 +146,80 @@ subtest '[qesap_az_vnet_peering_delete] delete failure' => sub {
     ok((any { /jsc#7487/ } @soft_failure), 'soft failure');
 };
 
+subtest '[qesap_az_get_tenant_id]' => sub {
+    my $qesap = Test::MockModule->new('qesapdeployment', no_auto => 1);
+    dies_ok { qesap_az_get_tenant_id() } 'Expected failure: missing mandatory arg';
+    my $input_value = 'Blackbeard';
+    $qesap->redefine(script_output => sub { return $input_value; });
+    dies_ok { qesap_az_get_tenant_id($input_value) } 'Expected failure: Returned value is not a valid UUID format';
+
+    $input_value = 'c0ffeeee-c0ff-eeee-1234-123456abcdef';
+    $qesap->redefine(script_output => sub { return $input_value; });
+    is qesap_az_get_tenant_id($input_value), 'c0ffeeee-c0ff-eeee-1234-123456abcdef', 'Returned value is a valid UUID';
```

Comment:
```suggestion
    is qesap_az_get_tenant_id($input_value), $input_value, 'Returned value is a valid UUID';
```

but at this point I'm confused about what the tested function does:
- you give an input `AAAA`
- it calls script_output that return `AAAA`
- the function return `AAAA`

Looking at the tested function code, the behavior is more like:
- you call the function with an input `AAAA`
- the function compose a cmd with `AAAA` in it and call script_output with it
- the function return whatever is returned by script_output (like `BBBB` not to confuse the reader)


```suggestion
subtest '[qesap_az_get_tenant_id]' => sub {
    my $qesap = Test::MockModule->new('qesapdeployment', no_auto => 1);
    my @calls;
    my $input_value = 'AAAAAAAA';
    my $output_value = 'BBBBBBBB';
    $qesap->redefine(script_output => sub { push @calls, $_[0]; return $output_value });

    my $result = qesap_az_get_tenant_id($input_value);

    ok((any { /az account show .*--subscription $input_value.*/ } @calls), 'az command properly composed');
    is $result, $output_value, 'Returned value is a valid UUID';
};
```

Reply:
if you like names less boring than `AAAA` and `BBBB` look at [here](https://pixar.fandom.com/wiki/Category:Finding_Nemo_Characters)

Reply:
Oh... what I meant here is similar to `qesap_az_enable_system_assigned_identity`

It's mostly about checking the regex inside qesap_az_get_tenant_id. So output value should be always the same as input value. Function should however croak if script_output result does not match the regex. 

https://github.com/os-autoinst/os-autoinst-distri-opensuse/compare/master...lpalovsky:os-autoinst-distri-opensuse:Azure-MSI-based-fencing#diff-4b8cb2b3db3a3b1a28d8774fe6e830e76c20a4c68e3e12bb862398ad9f4cc5b2R202

Looking at it now the unit tests are basically identical... You know what? Let me rewrite some stuff and come back. I can make it more simple. 
Thanks. That was a good point there.